modin/experimental/core/execution/native/implementations/hdk_on_native/dataframe/utils.py

```diff
@@ -13,11 +13,215 @@
 
 """Utilities for internal use by the ``HdkOnNativeDataframe``."""
 
-
+import re
 import pandas
 import pyarrow
 
+from pandas import Timestamp
 from modin.error_message import ErrorMessage
+from modin.utils import MODIN_UNNAMED_SERIES_LABEL
+from string import ascii_uppercase, ascii_lowercase, digits
+from typing import List, Tuple, Union
+
+IDX_COL_NAME = "__index__"
+ROWID_COL_NAME = "__rowid__"
+
+# Bytes 62 and 63 are encoded with 2 characters
+_BASE_EXT = ("_A", "_B")
+_BASE_LIST = tuple(ascii_uppercase + ascii_lowercase + digits) + _BASE_EXT
+_BASE_DICT = dict((c, i) for i, c in enumerate(_BASE_LIST))
+_NON_ALPHANUM_PATTERN = re.compile("[^a-zA-Z0-9]+")
+# Number of bytes in the tailing chunk
+_TAIL_LEN = {"_0": 0, "_1": 1, "_2": 2}
+_RESERVED_NAMES = (MODIN_UNNAMED_SERIES_LABEL, ROWID_COL_NAME)
+_COL_TYPES = Union[str, int, float, Timestamp, None]
+_COL_NAME_TYPE = Union[_COL_TYPES, Tuple[_COL_TYPES, ...]]
+
+
+def encode_col_name(
+    name: _COL_NAME_TYPE,
+    ignore_reserved: bool = True,
+) -> str:
+    """
+    Encode column name, using the alphanumeric and underscore characters only.
+
+    The supported name types are specified in the type hints. Non-string names
+    are converted to string and prefixed with a corresponding tag. The strings
+    are encoded in the following way:
+      - All alphanum characters are left as is. I.e., if the column name
+        consists from the alphanum characters only, the original name is
+        returned.
+      - Non-alphanum parts of the name are encoded, using a customized
+        version of the base64 algorithm, that allows alphanum characters only.
+
+    Parameters
+    ----------
+    name : str, int, float, Timestamp, None, tuple
+        Column name to be encoded.
+    ignore_reserved : bool, default: True
+        Do not encode reserved names.
+
+    Returns
+    -------
+    str
+        Encoded name.
+    """
+    if name is None:
+        return "_N"
+    if isinstance(name, int):
+        return f"_I{str(name)}"
+    if isinstance(name, float):
+        return f"_F{str(name)}"
+    if isinstance(name, Timestamp):
+        return f"_D{encode_col_name((name.timestamp(), str(name.tz)))}"
+    if isinstance(name, tuple):
+        dst = ["_T"]
+        count = len(name)
+        for n in name:
+            dst.append(encode_col_name(n))
+            count -= 1
+            if count != 0:
+                dst.append("_S")  # Separator
+        return "".join(dst)
+    if len(name) == 0:
+        return "_E"
+    if ignore_reserved and (name.startswith(IDX_COL_NAME) or name in _RESERVED_NAMES):
+        return name
+
+    non_alpha = _NON_ALPHANUM_PATTERN.search(name)
```

Comment:
why having non alpha-num characters is a problem for a column name? why we're getting rid of them?